x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/util/RemoteClusterRegistry.java

```diff
@@ -39,10 +37,11 @@ public RemoteClusterRegistry(RemoteClusterService remoteClusterService, IndicesO
         return incompatibleClusters;
     }
 
-    private Map<String, OriginalIndices> indicesPerRemoteCluster(String indexPattern) {
-        Map<String, OriginalIndices> indicesMap = remoteClusterService.groupIndices(indicesOptions,
-            Strings.splitStringByCommaToArray(indexPattern));
-        indicesMap.remove(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
-        return indicesMap;
+    public Set<String> clusterAliases(String[] indices, boolean discardLocal) {
```

Comment:
Does `discardLocal` has any other value than `true`? If not, better to keep the method specialized.

Reply:
> Does `discardLocal` has any other value than `true`?

Yes, the optimisation code needs to know if the queried indices are all local or not (so it calls with this parameter set to `false`).